airflow/providers/google/suite/hooks/calendar.py

```diff
@@ -0,0 +1,213 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+"""This module contains a Google Calendar API hook"""
+
+from datetime import datetime
+from typing import Any, Dict, Optional, Sequence, Union
+
+from googleapiclient.discovery import build
+
+from airflow.exceptions import AirflowException
+from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
+
+
+class GCalendarHook(GoogleBaseHook):
+    """
+    Interact with Google Calendar via Google Cloud connection
+    Reading and writing cells in Google Sheet:
+    https://developers.google.com/calendar/api/v3/reference
+
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param api_version: API Version
+    :type api_version: str
+    :param delegate_to: The account to impersonate using domain-wide delegation of authority,
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account.
+    :type impersonation_chain: Union[str, Sequence[str]]
+    """
+
+    def __init__(
+        self,
+        gcp_conn_id: str = 'google_cloud_default',
+        api_version: str = 'v3',
```

Comment:
Must we specify default value of our own? setting default value as such means that we will have to do a breaking change when v3 is deprecated.
We had similar case for facebook: https://github.com/apache/airflow/pull/18883

Reply:
I think this is a good point. In others Google operators we specify default version. But indeed I think we should make this a required parameter. WDYT @potiuk @mhenc ?

Reply:
@rsg17 what do you think?

Reply:
I like the idea of not specifying API versions to avoid the breaking change. I have one question: how would it work if the parameters required for the actual API call changed between versions?

I compared the [FB Ads API](https://developers.facebook.com/docs/marketing-api/insights/parameters/v12.0) with [Google Calendar](https://developers.google.com/calendar/api/v3/reference/events/list).

- I think the [FB Ads API](https://developers.facebook.com/docs/marketing-api/insights/parameters/v12.0) is much more structured with just 3 input parameters - account_id, params (query parameters for the report), fields (list of fields for the report).
- Compared with that, the [Google Calendar API](https://developers.google.com/calendar/api/v3/reference/events/list) has a lot of parameters.

Reply:
@turbaszek, @eladkal - I updated the code to remove our default value setting for api_version. This will be a required parameter (cannot be optional as the API requires it).